src/plugins/workspace/server/workspace_client.ts

```diff
@@ -125,6 +227,52 @@ export class WorkspaceClientWithSavedObject implements IWorkspaceDBImpl {
           type: WORKSPACE_TYPE,
         }
       );
+      const scopedClientWithoutPermissionCheck = this.getScopedClientWithoutPermission(requestDetail);
+      const tasks: Array<Promise<unknown>> = [];
+
+      /**
+       * Setup public workspace if public workspace can not be found
+       */
+      const hasPublicWorkspace = savedObjects.find((item) => item.id === PUBLIC_WORKSPACE_ID);
+
+      if (!hasPublicWorkspace) {
+        tasks.push(this.setupPublicWorkspace(scopedClientWithoutPermissionCheck));
+      }
+
+      /**
+       * Setup management workspace if management workspace can not be found
+       */
+      const hasManagementWorkspace = savedObjects.find(
+        (item) => item.id === MANAGEMENT_WORKSPACE_ID
+      );
+      if (!hasManagementWorkspace) {
+        tasks.push(this.setupManagementWorkspace(scopedClientWithoutPermissionCheck));
+      }
+
+      /**
+       * Setup personal workspace
+       */
+      const principals = SavedObjectsPermissionControl.getPrincipalsFromRequest(
+        requestDetail.request
+      );
+      /**
+       * Only when authentication is enabled will personal workspace be created
+       */
+      if (principals.users) {
+        const hasPersonalWorkspace = savedObjects.find((item) =>
+          principals.users?.includes(item.id)
```

Comment:
Can you leave a comment and say something like: user id will be personal workspace id

Reply:
Sure, done.